src/plot_api/plot_api.js

```diff
@@ -107,7 +107,7 @@ Plotly.plot = function(gd, data, layout, config) {
     // any part of the plotting code can push to gd._promises, then
     // before we move to the next step, we check that they're all
     // complete, and empty out the promise list again.
-    gd._promises = [];
+    if(!Array.isArray(gd._promises)) gd._promises = [];
```

Comment:
This I didn't write a test for... but I made this change so that if you make a plot, then change it before its promises have finished, you should still wait for them (via [`previousPromises`](https://github.com/plotly/plotly.js/pull/1991/files#diff-2941ab69a12080c0633ff4ac8ea3aa83R376)) so you don't get a race condition between them and the new plot (with the old ones potentially overwriting elements of the new plot you made).

I haven't actually managed to cause any ill effects without this change in manual testing, but I think it's safer this way.